src/main/java/frc/robot/subsystems/shooter/Shooter.java

```diff
@@ -0,0 +1,232 @@
+package frc.robot.subsystems.shooter;
+
+import static frc.robot.subsystems.shooter.ShooterConstants.*;
+
+import edu.wpi.first.math.interpolation.InterpolatingDoubleTreeMap;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.DriverStation.Alliance;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.lib.team3061.drivetrain.Drivetrain;
+import frc.lib.team3061.util.RobotOdometry;
+import frc.lib.team6328.util.TunableNumber;
+import frc.robot.Field2d;
+import frc.robot.subsystems.intake.Intake;
+import java.util.function.BooleanSupplier;
+import org.littletonrobotics.junction.Logger;
+
+public class Shooter extends SubsystemBase {
+
+  private ShooterIO io;
+  private Intake intake;
+  private Alliance alliance = Alliance.Red;
+  private InterpolatingDoubleTreeMap angleTreeMap;
+  private InterpolatingDoubleTreeMap veloTreeMap;
+  private final ShooterIOInputsAutoLogged shooterInputs = new ShooterIOInputsAutoLogged();
+
+  private final TunableNumber topWheelVelocity = new TunableNumber("Shooter/Top Wheel Velocity", 0);
+  private final TunableNumber bottomWheelVelocity =
+      new TunableNumber("Shooter/Bottom Wheel Velocity", 0);
+  private final TunableNumber angle = new TunableNumber("Shooter/Angle", 10.0);
+
+  public BooleanSupplier inStorage = () -> false;
+
+  private boolean autoShooter = false;
+  private boolean hasNote = false;
+  private boolean isAimed = false;
+  private boolean aiming = false;
+  private boolean scoringAmp = false;
+  private boolean scoringSubwoofer = false;
+  private boolean scoringPodium = false;
+
+  private int topAtSetpointIterationCount = 0;
+  private int bottomAtSetpointIterationCount = 0;
+  private int angleAtSetpointIterationCount = 0;
+
+  public Shooter(ShooterIO io, Drivetrain drivetrain, Intake intake) {
```

Comment:
I don't think the shooter needs a reference to the drivetrain; if I'm right, remove